hail/python/test/hail/linalg/test_linalg.py

```diff
@@ -37,6 +39,91 @@ def sparsify_numpy(np_mat, block_size, blocks_to_sparsify):
     return target_mat
 
 
+class batched_asserts(object):
+
+    def __init__(self, batch_size=32):
+        self._batch_size = batch_size
+
+    def __enter__(self):
+        self._a_list = []
+        self._b_list = []
+        self._modes = []
+        self._tbs = []
+        return self
+    
+    def _assert_agree(self, a, b, mode):
+        self._a_list.append(a)
+        self._b_list.append(b)
+        self._modes.append(mode)
+        self._tbs.append(traceback.format_stack())
+    def assert_eq(self, a, b):
+        self._assert_agree(a, b, 'eq')
+        
+    def assert_close(self, a, b):
+        self._assert_agree(a, b, 'close')
+
+    def __exit__(self, *exc):
+        a_list = self._a_list
+        b_list = self._b_list
+        modes = self._modes
+        tb = self._tbs
+        assert len(a_list) == len(b_list)
+        assert len(a_list) == len(modes)
+        assert len(a_list) == len(tb)
+
+        all_bms = {}
+
+        a_results = []
+        for i, a in enumerate(a_list):
+            if isinstance(a, BlockMatrix):
+                all_bms[(0, i)] = a.to_ndarray()
+                a_results.append(None)
+            else:
+                a_results.append(np.array(a))
+
+        b_results = []
+        for i, b in enumerate(b_list):
+            if isinstance(b, BlockMatrix):
+                all_bms[(1, i)] = b.to_ndarray()
+                b_results.append(None)
+            else:
+                b_results.append(np.array(b))
+
+        bm_keys = list(all_bms.keys())
+
+
+        vals = []
+        batch_size = self._batch_size
+        n_bms = len(bm_keys)
+        for batch_start in range(0, n_bms, batch_size):
+            vals.extend(list(hl.eval(tuple([all_bms[k] for k in bm_keys[batch_start:batch_start + batch_size]]))))
+
+        for (a_or_b, idx), v in zip(bm_keys, vals):
+            if a_or_b == 0:
+                a_results[idx] = v
+            else:
+                b_results[idx] = v
+
+        for i, x in enumerate(a_results):
+            assert x is not None, i
+        for i, x in enumerate(b_results):
+            assert x is not None, i
+
+        for a_res, b_res, mode, tb in zip(a_results, b_results, modes, tb):
+            print(a_res, b_res, mode, tb)
+            try:
+                if mode == 'close':
+                    assert np.allclose(a_res, b_res)
```

Comment:
Can we ust `np.testing.assert_allclose`? Could then just use the functions themselves instead of the `mode` strings.